src/processors/AsyncAPIInputProcessor.ts

```diff
@@ -0,0 +1,57 @@
+import { CommonInputModel } from '../models/CommonInputModel';
+import { AbstractInputProcessor } from '../models/AbstractInputProcessor';
+import {parse, AsyncAPIDocument, Schema as AsyncAPISchema} from '@asyncapi/parser';
+import { JsonSchemaInputProcessor } from './JsonSchemaInputProcessor';
+export class AsyncAPIInputProcessor extends AbstractInputProcessor {
+    /**
+     * Process the input as an AsyncAPI document
+     * 
+     * @param input 
+     */
+    async process(input: any): Promise<CommonInputModel> {
+        var doc: AsyncAPIDocument;
+        if(!AsyncAPIInputProcessor.isFromParser(input)){
+            doc = await parse(input);
+        }else{
+            doc = input;
+        }
+        let common = new CommonInputModel();
+        common.originalInput= doc;
```

Comment:
I think that `originalInput` should be `input` not parsed doc, but I understand why you did it in this way.

EDIT: Ok, we can leave it in this shape.

Reply:
Yea you are right... But `originalInput` is before simplification step, but yea lets keep it as is for now.